### DISCUSSION TITLE
Is _numpyconfig.h in packages/numpy/config still used?

### DISCUSSION THREAD

**Discussion by allsey87** [pyodide/pyodide#4943] [Q&A] [answered]
I have been working on compiling NumPy separately for a larger project and have been using Pyodide as a guide for how to do that. It seems to be that the vendored `_numpyconfig.h` was for an older version of NumPy and is no longer needed. I first noticed this because I think this file is missing `NPY_ABI_VERSION` and `NPY_API_VERSION`. I also tried removing this file and removing the following section from the `meta.yml` file and everything seems to at least build without any issues.
```yaml
cross-build-files:
  - numpy/core/include/numpy/numpyconfig.h
  - numpy/core/include/numpy/_numpyconfig.h
  - numpy/core/lib/libnpymath.a
  - numpy/random/lib/libnpyrandom.a
```

**Comment by hoodmane** [comment, marked as answer]
I think `config/_numpyconfig.h` hasn't been used since:
https://github.com/numpy/numpy/commit/3e4d87b1ba76b245dcbf45bb26b1f7118861f286

The `cross-build-files` is important for all packages that use numpy at build time.

**Reply by allsey87** [reply]
Right, so `cross-build-files` is about what is exported to other packages not how the current package is being built.

**Reply by hoodmane** [reply]
Exactly. When another package says it needs `numpy` to build, we download the numpy manylinux wheel and then copy the cross-build-files from the wasm wheel into the manylinux wheels.
